### PR TITLE
Add kinematic-only feature for fixed non-articulated objects for better performance

### PR DESCRIPTION
**Pull request by cremebrule**

### CODE REVIEW

File: omnigibson/objects/primitive_object.py [hunk 1]
```diff
@@ -165,8 +165,9 @@ def _post_load(self):
         visual_geom_prim.color = self._load_config["color"]
         visual_geom_prim.opacity = self._load_config["opacity"]
 
-        # Update collision approximation
-        self.root_link.collision_meshes["collision"].set_collision_approximation("convexHull")
+        # Set the collision approximation appropriately
+        col_approximation = "boundingSphere" if self._primitive_type == "Sphere" else "convexHull"
```
Comment (ChengshuLi) [inline omnigibson/objects/primitive_object.py:169]: also do it for box

File: omnigibson/prims/geom_prim.py [hunk 2]
```diff
@@ -261,11 +263,28 @@ def get_min_torsional_patch_radius(self):
...
-            approximation_type (str): approximation used for collision, could be "none", "convexHull" or "convexDecomposition"
+            approximation_type (str or None): approximation used for collision.
+                Can be one of: {"none", "convexHull", "convexDecomposition", "meshSimplification", "sdf"}
+                If None, the approximation will use the underlying triangle mesh.
         """
         assert self._mesh_collision_api is not None, "collision_approximation only applicable for meshes!"
+
+        # Make sure to add the appropriate API if we're setting certain values
+        if approximation_type == "convexHull" and not self._prim.HasAPI(PhysxSchema.PhysxConvexHullCollisionAPI):
+            PhysxSchema.PhysxConvexHullCollisionAPI.Apply(self._prim)
```
Comment (ChengshuLi) [inline omnigibson/prims/geom_prim.py:274]: add 60 vertices upper bound for gpu

File: omnigibson/prims/geom_prim.py [hunk 3]
```diff
@@ -261,11 +263,28 @@ def get_min_torsional_patch_radius(self):
...
+        elif approximation_type == "convexDecomposition" and not self._prim.HasAPI(PhysxSchema.PhysxConvexDecompositionCollisionAPI):
+            PhysxSchema.PhysxConvexDecompositionCollisionAPI.Apply(self._prim)
+        elif approximation_type == "meshSimplification" and not self._prim.HasAPI(PhysxSchema.PhysxTriangleMeshSimplificationCollisionAPI):
+            PhysxSchema.PhysxTriangleMeshSimplificationCollisionAPI.Apply(self._prim)
+        elif approximation_type == "sdf" and not self._prim.HasAPI(PhysxSchema.PhysxSDFMeshCollisionAPI):
+            PhysxSchema.PhysxSDFMeshCollisionAPI.Apply(self._prim)
+        elif approximation_type in {"none", None} and not self._prim.HasAPI(PhysxSchema.PhysxTriangleMeshCollisionAPI):
+            PhysxSchema.PhysxTriangleMeshCollisionAPI.Apply(self._prim)
+        else:
+            raise ValueError(f"Got invalid collision approximation type: {approximation_type}. "
```
Comment (ChengshuLi) [inline omnigibson/prims/geom_prim.py:284]: skip boundSphere and boundBox?

**Review by ChengshuLi — APPROVED** [review]

File: omnigibson/objects/primitive_object.py (hunk 1 above)
Comment (cremebrule) [reply omnigibson/objects/primitive_object.py:169]: done!

File: omnigibson/prims/geom_prim.py (hunk 2 above)
Comment (cremebrule) [reply omnigibson/prims/geom_prim.py:274]: done!

File: omnigibson/prims/geom_prim.py (hunk 3 above)
Comment (cremebrule) [reply omnigibson/prims/geom_prim.py:284]: done!

**Pull request merged**